test/fixtures/cookbooks/mysql2_chef_gem_test/recipes/default.rb

```diff
@@ -9,9 +9,14 @@
   include_recipe 'yum-mysql-community::mysql57'
 end
 
-mysql2_chef_gem 'default' do
-  package_version node['mysql2_chef_gem']['package_version'] if node['mysql2_chef_gem']
-  provider Chef::Provider::Mysql2ChefGem::Mysql if node['mysql2_chef_gem']['provider'] == 'mysql'
-  provider Chef::Provider::Mysql2ChefGem::Mariadb if node['mysql2_chef_gem']['provider'] == 'mariadb'
-  action :install
+if node['mysql2_chef_gem']['provider'] == 'mysql'
```

Comment:
I fixed the spec, but it could use a refactor as it's really testing nothing, but the matcher right now